test/WebJobs.Script.Tests/Environment/EnvironmentTests.cs

```diff
@@ -0,0 +1,129 @@
+            Assert.False(environment.IsLinuxAppServiceEnvWithPersistentStorage());
+        }
+
+        [Fact]
+        public void IsLinuxAppServiceEnvWithPersistentStorage_NotLinuxWithStorageSet_ReturnsFalse()
+        {
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(AzureWebsiteInstanceId, Guid.NewGuid().ToString("N"));
+            environment.SetEnvironmentVariable(AzureAppServiceStorage, "true");
+            Assert.False(environment.IsLinuxAppServiceEnvWithPersistentStorage());
+        }
+
+        [Fact]
+        public void IsLinuxAppServiceEnvWithPersistentStorage_StorageSetToTrue_ReturnsTrue()
+        {
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(AzureWebsiteInstanceId, Guid.NewGuid().ToString("N"));
+            environment.SetEnvironmentVariable(FunctionsLogsMountPath, Guid.NewGuid().ToString("N"));
+            environment.SetEnvironmentVariable(AzureAppServiceStorage, "true");
+            Assert.True(environment.IsLinuxAppServiceEnvWithPersistentStorage());
+        }
+
+        [Fact]
+        public void IsPersistentStorageAvailable_IsWindows_ReturnsTrue()
+        {
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(AzureWebsiteInstanceId, Guid.NewGuid().ToString("N"));
+            Assert.True(environment.IsAppServiceWindowsEnvironment());
+            Assert.True(environment.IsPersistentFileSystemAvailable());
+        }
+
+        [Fact]
+        public void IsPersistentStorageAvailable_IsCoreTools_ReturnsTrue()
+        {
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(CoreToolsEnvironment, "true");
+            Assert.True(environment.IsPersistentFileSystemAvailable());
+        }
+
+        [Fact]
+        public void IsPersistentStorageAvailable_IsLinuxWithoutStorage_ReturnsFalse()
+        {
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(AzureWebsiteInstanceId, Guid.NewGuid().ToString("N"));
+            environment.SetEnvironmentVariable(FunctionsLogsMountPath, Guid.NewGuid().ToString("N"));
+            environment.SetEnvironmentVariable(AzureAppServiceStorage, "false");
+            Assert.False(environment.IsLinuxAppServiceEnvWithPersistentStorage());
+            Assert.False(environment.IsPersistentFileSystemAvailable());
+        }
+
+        [Fact]
+        public void IsPersistentStorageAvailable_IsLinuxWithStorage_ReturnsTrue()
+        {
+            var environment = new TestEnvironment();
+            environment.SetEnvironmentVariable(AzureWebsiteInstanceId, Guid.NewGuid().ToString("N"));
+            environment.SetEnvironmentVariable(FunctionsLogsMountPath, Guid.NewGuid().ToString("N"));
+            Assert.True(environment.IsLinuxAppServiceEnvWithPersistentStorage());
```

Comment:
Same here question here:  IsPersistentStorageAvailable returns true even though App setting is not set. Is this the expected behavior?

Reply:
Yes